File: samples/SampleCounter/CounterPresenter.cs
namespace SampleCounter;

public sealed class CounterPresenter
{
    private readonly ICounterView _view;
    private Counter _counter = Counter.Zero;

    public CounterPresenter(ICounterView view)
    {
        _view = view;

        Initialize();
    }

    private void Initialize()
    {
        _view.DisplayCounter(_counter);
    }

    public void Increment()
    {
        _counter = _counter.Increment();
        DisplayCounter();
    }

    public void Decrement()
    {
        _counter = _counter.Decrement();
        DisplayCounter();
    }

    private void DisplayCounter()
    {
        _view.DisplayCounter(_counter);
    }
}

File: targets/js/sample-counter/lib/src/sample-counter/counter-presenter.js
import { Counter } from "./counter";
export class CounterPresenter {
    _view;
    _counter = Counter.zero;
    constructor(view) {
        this._view = view;
        this.initialize();
    }
    initialize() {
        this._view.displayCounter(this._counter);
    }
    increment() {
        this._counter = this._counter.increment();
        this.displayCounter();
    }
    decrement() {
        this._counter = this._counter.decrement();
        this.displayCounter();
    }
    displayCounter() {
        this._view.displayCounter(this._counter);
    }
}
